### ISSUE TITLE
Flask plugin breaks DeclarativeBase

### ISSUE THREAD

**Issue by pvanderlinden** [pallets-eco/flask-sqlalchemy#1361]
The plugin breaks the DeclarativeBase, e.g.:

```
from typing import Any

from flask_sqlalchemy import SQLAlchemy
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.types import JSON

class Base(DeclarativeBase):
    type_annotation_map = {dict[str, Any]: JSON}
db = SQLAlchemy(model_class=Base)
```

Gives the error:
`sqlalchemy.exc.InvalidRequestError: Declarative base class has both a 'registry' attribute and a type_annotation_map entry.  Per-base type_annotation_maps are not supported.  Please apply the type_annotation_map to this registry directly.`

However, `type_annotation_map` is supported on the base, but by subclassing the base class (something which seems to be happening inside the plugin), it now detects both the registry created from the base class with `type_annotation_map` and the `type_annotation_map`. This doesn't happen without the Flask plugin.

Environment:

- Python version: 3.11.6
- Flask-SQLAlchemy version: 3.1.1
- SQLAlchemy version: 2.0.31

**Comment by samuelhwilliams** [comment]
Yep, I get the same - unsure what the right way around this is :(

**Comment by pamelafox** [comment]
Are you able to use Flask-SQLAlchemy-Lite for your use case? https://flask-sqlalchemy-lite.readthedocs.io/en/latest/

**Comment by samuelhwilliams** [comment]
Theoretically yes - but if that's the official advice it's probably be good to hear that and see this deprecated? If this is still being maintained then it feels like there should be an answer to this question.

Switching to FSL is probably not a hard migration, but it's a chunk of work obviously still. And FSL is a very young lib in comparison.

**Comment by Slyces** [comment]
Hello everyone. I stumbled upon this issue myself, and found the following workaround (snippet from my codebase):
```python
db = SQLAlchemy()
db.Model.registry.update_type_annotation_map(
    {
        datetime: sa.DateTime(timezone=True),
        dict[str, Any]: JSONB().with_variant(sa.JSON, "sqlite"),
        uuid.UUID: UUID(as_uuid=True).with_variant(SQLiteUUID(), "sqlite"),
    }
)
```
This seems to solve the issue for me, hope it will help if anyone else is having this issue :)
